packages/babel-preset-fbjs/plugins/inline-requires.js

```diff
@@ -235,7 +249,21 @@ function getInlineableModule(node, state) {
         : null;
   }
 
-  return moduleName == null || state.ignoredRequires.has(moduleName)
+  // Check if require is in any parent scope
+  let parentPath = path.parentPath;
+  let isRequireInScope = false;
+  const fnName = node.callee.name;
+  while (parentPath) {
+    if (parentPath.scope.getBinding(fnName) != null) {
+      isRequireInScope = true;
+      break;
+    }
+    parentPath = parentPath.parentPath;
+  }
```

Comment:
You don't have to traverse up in the tree. `scope.getBinding()` checks the current scope and any parent scope

https://github.com/babel/babel/blob/master/packages/babel-traverse/src/scope/index.js#L993-L1017

Reply:
Thank you. Updated